bot/skills/length.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from typing import Optional, List, TypedDict
+
+from telegram import Update, User, Message, ParseMode
+from telegram.ext import Updater, CommandHandler, CallbackContext
+from telegram.ext.filters import Filters
+from telegram.utils.helpers import mention_markdown
+
+from mode import cleanup_queue_update
+from config import get_group_chat_id
+
+from skills.roll import _get_username
+
+from db.mongo import get_db
+import pymongo
+from pymongo.collection import Collection
+from pymongo.results import UpdateResult
+
+logger = logging.getLogger(__name__)
+
+PENINSULA_LENGTH_FIELD = "peninsula_length"
+PENINSULA_MAXIMAL_LENGTH = 20
+
+
+class PeninsulaDataType(TypedDict):
+    _id: str
+    peninsula_length: int
+    meta: User
+
+
+def add_length(upd: Updater, handlers_group: int):
+    logger.info("registering length handlers")
+    dp = upd.dispatcher
+    dp.add_handler(
+        CommandHandler(
+            "length",
+            _length,
+            filters=Filters.chat(username=get_group_chat_id().strip("@")),
+            run_async=True,
+        ),
+        handlers_group,
+    )
+
+    dp.add_handler(
+        CommandHandler(
+            "longest",
+            _longest,
+            filters=Filters.chat(username=get_group_chat_id().strip("@")),
+            run_async=True,
+        ),
+        handlers_group,
+    )
+
+
+class DB:
+    def __init__(self, db_name: str):
+        self._coll: Collection = get_db(db_name).peninsulas
+
+    def get_longest_peninsula(self) -> int:
+        result = PENINSULA_MAXIMAL_LENGTH
+        x: PeninsulaDataType = self._coll.find_one(
+            sort=[(PENINSULA_LENGTH_FIELD, pymongo.DESCENDING)]
+        )
+        if x is not None:
+            result = max(result, int(x[PENINSULA_LENGTH_FIELD]))
```

Comment:
Wouldn't this return 20 every time?

Reply:
Yes, until telegram id would exceed 20 symbols. Actually that magic number was taken to limit eggplant amount to fit in one line on pc, we can probably tweak it smaller to fit on mobile.